### ISSUE TITLE
[Q] Get rid of String class

### ISSUE THREAD

**Issue by DaeMonSxy** [mathieucarbou/ESPAsyncWebServer#83]
How about to get rid of String class? 
outdated, unstable etc
move the whole lib to char array, or std::string ( there are benefits)

**Comment by mathieucarbou** [comment]
No, this would be a breaking change. Also, this is a arduino lib and String is used pretty much everywhere.

But if you feel to raise a PR with some macro definitions to swap the String implementation used, please go ahead.
